crates/orderbook/src/api.rs

```diff
@@ -75,12 +75,19 @@ pub fn handle_all_routes(
         ("v1/post_quote", box_filter(post_quote::post_quote(quotes))),
         (
             "v1/auction",
-            box_filter(get_auction::get_auction(orderbook)),
+            box_filter(get_auction::get_auction(orderbook.clone())),
         ),
         (
             "v1/solver_competition",
             box_filter(get_solver_competition::get(Arc::new(database.clone()))),
         ),
+        (
+            "v1/solver_competition",
```

Comment:
This identifier is used for metrics so it should be unique to not interfere with the metrics for the existing endpoint.

Reply:
it is actually used for the solver_competition/auction_id and solver_competition/by_tx_hash as well which seem to work. so it should work for the newly added endpoint as well, right ?

Reply:
Ahh, right. Didn't see the additional occurrence.
I think this is a bug and we should make them unique. Especially the one route below is very different from the one you are implementing.